text/php/0002-buildpack-yml-to-env-var.md

```diff
@@ -0,0 +1,222 @@
+# Buildpack.yml to Environment Variables
+
+## Proposal
+
+Migrate to using environment variables to do all buildpack configuration and
+get rid of `buildpack.yml`.
+
+## Motivation
+
+There are several reasons for making this switch.
+1. There is already an [existing RFC](https://github.com/paketo-buildpacks/rfcs/blob/main/text/0026-environment-variable-configuration-of-buildpacks.md)
+   that proposes moving away from `buildpack.yml` as a configuration tool.
+1. Environment variables appears to be the standard for configuration in other
+   buildpack ecosystems such as Google Buildpacks and Heroku as well as the
+   Paketo Java buildpacks. Making this change will align the buildpack with the
+   rest of the buildpack ecosystem.
+1. There is native support to pass environment variables to the buildpack
+   either on a per run basis or by configuration that can be checked into
+   source control, in the form of `project.toml`.
+
+## Implementation
+
+The proposed environment variables for PHP Web are as follows:
+
+### `BP_PHP_VERSION`
+```shell
+$BP_PHP_VERSION="7.4.x"
+```
+
+This will replace the following structure in `buildpack.yml`:
+```yaml
+php:
+ version: 7.4.x
+```
+
+### `BP_PHP_SERVER`
+```shell
+$BP_PHP_SERVER="php-server"
+```
+
+This will replace the following structure in `buildpack.yml`:
+```yaml
+php:
+ webserver: php-server
+```
+### `BP_PHP_WEB_DIR`
+```shell
+$BP_PHP_WEB_DIR="htdocs"
+```
+
+This will replace the following structure in `buildpack.yml`:
+```yaml
+php:
+ webdirectory: htdocs
+```
+
+### `BP_PHP_LIB_DIR`
+```shell
+$BP_PHP_LIB_DIR="lib"
+```
+
+This will replace the following structure in `buildpack.yml`:
+```yaml
+php:
+ libdirectory: lib
+```
+
+### `BP_PHP_SERVER_ADMIN`
+```shell
+$BP_PHP_SERVER_ADMIN="admin@localhost"
+```
+
+This will replace the following structure in `buildpack.yml`:
+```yaml
+php:
+ serveradmin: admin@localhost
+```
+
+### `BP_REDIS_SESSION_STORE_SERVICE_NAME`
+```shell
+$BP_REDIS_SESSION_STORE_SERVICE_NAME="redis-session"
+```
+
+This will replace the following structure in `buildpack.yml`:
+```yaml
+php:
+ redis:
+   session_store_service_name: redis-sessions
+```
+
+### `BP_MEMCACHED_SESSION_STORE_SERVICE_NAME`
+```shell
+$BP_MEMCACHED_SESSION_STORE_SERVICE_NAME="memcached-session"
+```
+
+This will replace the following structure in `buildpack.yml`:
+```yaml
+php:
+ memcached:
+   session_store_service_name: memcached-sessions
+```
```

Comment:
I suspect that we can do away with these. Right now, it's being used to specify the name of the service binding, but we should probably just switch this to a documented service binding type. Thus you'd have a type like `PHPRedisSessions` and `PHPMemcachedSessions` that would be fixed. The user would then create a binding with this type if they want to enable this functionality. The binding name is then irrelevant and you don't need a configuration option for it.

Reply:
I am more than happy with that solution! I will make the proper changes.

Reply:
Type names have been proposed in the removed configuration section